CMake/CheckAtomic.cmake

```diff
@@ -70,9 +70,11 @@ if(NOT HAVE_CXX_ATOMICS64_WITHOUT_LIB)
   check_library_exists(atomic __atomic_load_8 "" HAVE_CXX_LIBATOMICS64)
   if(HAVE_CXX_LIBATOMICS64)
     list(APPEND CMAKE_REQUIRED_LIBRARIES "atomic")
-    check_working_cxx_atomics64(HAVE_CXX_ATOMICS64_WITH_LIB)
-    if (NOT HAVE_CXX_ATOMICS64_WITH_LIB)
-      message(FATAL_ERROR "Host compiler must support std::atomic!")
+    if (NOT CMAKE_CROSSCOMPILING)
+      check_working_cxx_atomics64(HAVE_CXX_ATOMICS64_WITH_LIB)
```

Comment:
because the check, in itself, is trying to compile and run the code, there is no way to run an ARM binary on x86, so the check will fail, even if the atomic is present

Reply:
The check was not originally about whether `std::atomic` was supported, though, but instead whether `-latomic` is necessary for compiling and/or linking.  Also, the check only compiles, and doesn't run, so I don't think cross-compiling should be a problem.

This code was originally added fairly recently: https://github.com/mlpack/mlpack/pull/3153/files
and so I'm pretty hesitant to remove it entirely and potentially break the Debian package again.